docs/onboarding.rst

```diff
@@ -328,3 +328,8 @@ What is MS/PFS?
 How are messages encoded?
   The `messages specification <https://raiden-network-specification.readthedocs.io/en/latest/messaging.html>`_ should describe the message formats.  On the implementation side, the `messages file <https://github.com/raiden-network/raiden/blob/4979c27f5d031dd44963f7f7a4b10a7d20cee712/raiden/messages.py>`_ contains the details.  See ``encode()`` method of ``Message`` class, and ``packed()`` method of each subclass of ``Message``.  Ultimately the byte sequence of the packed messages is defined in the `encoding/messages file <https://github.com/raiden-network/raiden/blob/a19a6c853b55f13725f2545c77b0475cbcc86807/raiden/encoding/messages.py#L58>`_.
 
+Why does `updateNonClosingBalanceProof() <https://github.com/raiden-network/raiden-contracts/blob/04fe5a6c33b2a34cd893c5e546b8df949e194947/raiden_contracts/contracts/TokenNetwork.sol#L536>`_ require two signatures?
+  The closing participant's signature is required because the balance proof here is sent from the closing participant to the non-closing participant. On this direction of the channel, the closing participant has control over issuing balance proofs. On the other side, the non-closing participant has a choice of accepting or refusing the balance proofs. For example, when the transferred amount decreases in a new balance proof, the non-closing participant should refuse the balance proof. In such cases, the non-closing participant should never add the second signature on the balance proof.
```

Comment:
Hmm I am not sure I understand this the way it is currently worded here @pirapira. If we call `updateNonClosingBalanceProof()` the `closing_signature` is indeed to update the closer's side of the channel. What he has sent to us (the non-closing participant). The non-closing signature is essentially our signature and the only reason we need it for is so that a third party can call the function with our signature for us to register the closing participant's BP onchain.

Reply:
> so that a third party can call the function

We can drop `non_closing_signature` from the argument list, and still, any third party can call `updateNonClosingBalanceProof()`.

Reply:
I think @LefterisJP assumes that we check the caller of the function instead of verifying the `non_closing_signature`. If we do that, his statement makes a lot of sense. If we don't add that check instead, I agree with @pirapira .

Reply:
I think I'm hearing two questions:
* why do `closeChannel()` and `updateNonClosingBalanceProof()` have different number of signatures in the arguments
* why do these functions require signatures from both sides.

I'll split the question into two.